File: iOS/DataPool/Sources/DataPool/Application/CreatePoolUseCase.swift
public class CreatePoolUseCase {
    private let poolRepository: PoolRepository

    public init(poolRepository: PoolRepository) {
        self.poolRepository = poolRepository
    }

    public func execute(createPoolInput: CreatePoolInput) async -> Result<CreatePoolOutput, Error> {
        await poolRepository.createPool(createPoolInput: createPoolInput)
    }
}

File: Android/data/pool/src/main/java/com/felipearpa/tyche/data/pool/application/CreatePoolUseCase.kt
package com.felipearpa.tyche.data.pool.application

import com.felipearpa.tyche.data.pool.domain.CreatePoolInput
import com.felipearpa.tyche.data.pool.domain.PoolRepository

class CreatePoolUseCase internal constructor(private val poolRepository: PoolRepository) {
    suspend fun execute(createPoolInput: CreatePoolInput) =
        poolRepository.createPool(createPoolInput = createPoolInput)
}
